app/src/main/java/com/example/pygmyhippo/common/NotificationCenter.java

```diff
@@ -0,0 +1,200 @@
+package com.example.pygmyhippo.common;
+
+/*
+ * Handles notification logic for PygmHippo app.
+ *
+ * Implementation:
+ * A snapshot listener is installed onto the Events collection. Whenever the collection is modified,
+ * the snapshot listener will be invoked and all the events will be scanned for entrants with the
+ * same accountID as the accountID used in making NotificationCenter. If an entrant is found,
+ * and the entrantStatus != notifiedStatus, then a notification will be posted and the corresponding
+ * entrant's notifiedStatus will update to match the entrantStatus.
```

Comment:
just a bit confused, how exactly does this work?

Reply:
Which part are you confused about?

Reply:
yolo, approved 👍 (i dont really get whats happening here: 
![image](https://github.com/user-attachments/assets/41d3fe27-ebf2-4bc8-bd8a-864e3d8b2d53)
)